Bio/Seq.py

```diff
@@ -2052,6 +2052,22 @@ def ungap(self, gap="-"):
             raise ValueError(f"Unexpected gap character, {gap!r}")
         return self.replace(gap, b"")
 
+    @property
+    def defined(self):
+        """Return true if the sequence is defined, false if undefined or partially defined.
+
+        Zero-length sequences are always considered to be defined.
+        """
+        if len(self) == 0:
+            return True
+        elif isinstance(self, UnknownSeq):
```

Comment:
This can be turned into a single isinstance with a tuple of classes:

```python
elif isinstance(self, (UnknownSeq, _UndefinedSequenceData, _PartiallyDefinedSequenceData)):
```

Reply:
One of the checks is on `self`, the other is on `self._data`, but that's still one less branch 👍

Reply:
You can get rid of the `isinstance(self, UnknownSeq)` branch by implementing `defined` on the `UnknownSeq` class.